fatal error while looking into the details of a preview

### What happened?

Pulumi cli crashed while looking at the details of a preview:
================================================================================

```
Pulumi Version:   v3.43.1
Go Version:       go1.19.2
Go Compiler:      gc
Architecture:     amd64
Operating System: windows
Panic:            fatal: An assertion has failed

goroutine 1 [running]:
runtime/debug.Stack()
        /opt/hostedtoolcache/go/1.19.2/x64/src/runtime/debug/stack.go:24 +0x65
main.panicHandler()
        /home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/main.go:29 +0x3d
panic({0x2a626a0, 0xc00921d7f0})
        /opt/hostedtoolcache/go/1.19.2/x64/src/runtime/panic.go:884 +0x212
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
        /home/runner/work/pulumi/pulumi/sdk/go/common/util/contract/failfast.go:23
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.Assert(...)
        /home/runner/work/pulumi/pulumi/sdk/go/common/util/contract/assert.go:26
github.com/pulumi/pulumi/pkg/v3/backend/display.(*propertyPrinter).printArchiveDiff(0x2e49020?, 0xc001f053a8, 0xc007cc3ef0, 0xc007cc3f90)
        /home/runner/work/pulumi/pulumi/pkg/backend/display/object_diff.go:875 +0x8bc
github.com/pulumi/pulumi/pkg/v3/backend/display.(*propertyPrinter).printPropertyValueDiff(0x2b365c0?, 0xc001f053a8, {{{0x2e49020, 0xc007cc3ef0}}, {{0x2e49020, 0xc007cc3f90}}, 0x0, 0x0})
        /home/runner/work/pulumi/pulumi/pkg/backend/display/object_diff.go:768 +0x456
github.com/pulumi/pulumi/pkg/v3/backend/display.(*propertyPrinter).printObjectPropertyDiff(0xc0092018f0, {0xc002160db6, 0x6}, 0x6, {0xc009201800?, 0xc009201830?, 0xc009201860?, 0xc009201890?})
        /home/runner/work/pulumi/pulumi/pkg/backend/display/object_diff.go:721 +0x20c
github.com/pulumi/pulumi/pkg/v3/backend/display.(*propertyPrinter).printObjectDiff(0xc0092018f0, {0xc009201800, 0xc009201830, 0xc009201860, 0xc009201890}, {0x0, 0x0, 0x0?})
        /home/runner/work/pulumi/pulumi/pkg/backend/display/object_diff.go:710 +0x34f
github.com/pulumi/pulumi/pkg/v3/backend/display.PrintObjectDiff(0xc0092017d0, {0xc009201800?, 0xc009201830?, 0xc009201860?, 0xc009201890?}, {0x0, 0x0, 0x0}, 0x1, 0x3, ...)
        /home/runner/work/pulumi/pulumi/pkg/backend/display/object_diff.go:684 +0x191
github.com/pulumi/pulumi/pkg/v3/backend/display.renderDiff({_, _}, {{0x2eb7b2e, 0x7}, {0xc0072fe880, 0x7b}, {0xc00119c640, 0x19}, 0xc00846fef0, 0xc00921e000, ...}, ...)
        /home/runner/work/pulumi/pulumi/pkg/backend/display/diff.go:319 +0x2a5
github.com/pulumi/pulumi/pkg/v3/backend/display.renderDiffResourcePreEvent({{{0x2eb7b2e, 0x7}, {0xc0072fe880, 0x7b}, {0xc00119c640, 0x19}, 0xc00846fef0, 0xc00921e000, 0xc00921e090, {0xc00921d6c0, ...}, ...}, ...}, ...)
        /home/runner/work/pulumi/pulumi/pkg/backend/display/diff.go:347 +0x36d
github.com/pulumi/pulumi/pkg/v3/backend/display.RenderDiffEvent({{0x2ec3e3f?, 0x0?}, {0x2cc7520?, 0xc007645400?}}, 0x2a1b4c0?, {{0x2eb373e, 0x6}, 0x0, 0x0, 0x0, ...})
        /home/runner/work/pulumi/pulumi/pkg/backend/display/diff.go:119 +0x565
github.com/pulumi/pulumi/pkg/v3/backend.createDiff({0x0?, 0x1?}, {0xc006478000, 0xb7, 0x1?}, {{0x2eb373e, 0x6}, 0x0, 0x0, 0x0, ...})
        /home/runner/work/pulumi/pulumi/pkg/backend/apply.go:274 +0x26d
github.com/pulumi/pulumi/pkg/v3/backend.confirmBeforeUpdating({_, _}, {_, _}, {_, _, _}, {{{0x450fb30, 0x0, 0x0}, ...}, ...})
        /home/runner/work/pulumi/pulumi/pkg/backend/apply.go:196 +0x5aa
github.com/pulumi/pulumi/pkg/v3/backend.PreviewThenPrompt({_, _}, {_, _}, {_, _}, {0xc0002e6280, {0xc000249040, 0x46}, {0x0, ...}, ...}, ...)
        /home/runner/work/pulumi/pulumi/pkg/backend/apply.go:137 +0x3b9
github.com/pulumi/pulumi/pkg/v3/backend.PreviewThenPromptThenExecute({_, _}, {_, _}, {_, _}, {0xc0002e6280, {0xc000249040, 0x46}, {0x0, ...}, ...}, ...)
        /home/runner/work/pulumi/pulumi/pkg/backend/apply.go:218 +0x13e
github.com/pulumi/pulumi/pkg/v3/backend/httpstate.(*cloudBackend).Update(_, {_, _}, {_, _}, {0xc0002e6280, {0xc000249040, 0x46}, {0x0, 0x0, ...}, ...})
        /home/runner/work/pulumi/pulumi/pkg/backend/httpstate/backend.go:956 +0xc5
github.com/pulumi/pulumi/pkg/v3/backend.UpdateStack({_, _}, {_, _}, {0xc0002e6280, {0xc000249040, 0x46}, {0x0, 0x0, 0x0}, ...})
        /home/runner/work/pulumi/pulumi/pkg/backend/stack.go:91 +0xba
github.com/pulumi/pulumi/pkg/v3/backend/httpstate.(*cloudStack).Update(_, {_, _}, {0xc0002e6280, {0xc000249040, 0x46}, {0x0, 0x0, 0x0}, 0xc0003dd4b8, ...})
        /home/runner/work/pulumi/pulumi/pkg/backend/httpstate/stack.go:152 +0x88
main.newUpCmd.func1({_, _}, {{{0x450fb30, 0x0, 0x0}, {0x0, 0x0, 0x0}, 0x7fffffff, 0x0, ...}, ...})
        /home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/up.go:188 +0xe17
main.newUpCmd.func3(0x0?, {0xc0004721d0, 0x0, 0x0?})
        /home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/up.go:468 +0x665
github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil.RunResultFunc.func1(0xc000a9aa00?, {0xc0004721d0, 0x0, 0x1})
        /home/runner/work/pulumi/pulumi/sdk/go/common/util/cmdutil/exit.go:112 +0x4f
github.com/spf13/cobra.(*Command).execute(0xc000a9aa00, {0xc0004721c0, 0x1, 0x1})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:876 +0x67b
github.com/spf13/cobra.(*Command).ExecuteC(0xc000a1ac80)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
        /home/runner/work/pulumi/pulumi/pkg/cmd/pulumi/main.go:48 +0x45
```

### Steps to reproduce

started with 
```
> pulumi up -r  
Do you want to perform this update? 
> details
```
### Expected Behavior

The details of resources that were going to be updated / replaced 

### Actual Behavior

fatal error, encounterd a panic.

### Output of `pulumi about`

```

running 'dotnet build -nologo .'
  Determining projects to restore...

  All projects are up-to-date for restore.

  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview

  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview

  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview

  SCloud.Core -> C:\_src\cloud\github\SearchCloudIAC\src\SCloud.Core\bin\Debug\net6.0\SCloud.Core.dll

  Application -> C:\_src\cloud\github\SearchCloudIAC\src\Application\bin\Debug\net6.0\Application.dll

  availability-check -> C:\_src\cloud\github\SearchCloudIAC\src\availability-check\bin\Debug\availability-check.dll

C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\CloudZone.cs(141,38): warning CS8602: Dereference of a possibly null reference. [C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\sinequa-cloudzone.csproj]

C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\CloudZone.cs(141,38): warning CS8619: Nullability of reference types in value of type 'Output<string?>' doesn't match target type 'Output<string>'. [C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\sinequa-cloudzone.csproj]
C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\CloudZone.cs(146,50): warning CS8601: Possible null reference assignment. [C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\sinequa-cloudzone.csproj]
C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\CloudZone.cs(147,51): warning CS8602: Dereference of a possibly null reference. [C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\sinequa-cloudzone.csproj]
C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\CloudZone.cs(63,16): warning CS8618: Non-nullable property 'AvailabilityCheckContainerUri' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\sinequa-cloudzone.csproj]
C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\CloudZone.cs(210,24): warning CS8603: Possible null reference return. [C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\sinequa-cloudzone.csproj]
C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\CloudZone.cs(215,24): warning CS8603: Possible null reference return. [C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\sinequa-cloudzone.csproj]
C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\CloudZone.cs(366,43): warning CS8619: Nullability of reference types in value of type 'Output<string?>' doesn't match target type 'Output<string>'. [C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\sinequa-cloudzone.csproj]

  sinequa-cloudzone -> C:\_src\cloud\github\SearchCloudIAC\src\PulumiStacks\sinequa-cloudzone\bin\Debug\net6.0\sinequa-cloudzone.dll



Build succeeded.
```
### Additional context

I'm guessing this is related to our stack, and can be hard to reproduce.  

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

I am getting a similar error.

This might be digital ocean specific.

after running

```
 DIGITALOCEAN_TOKEN=MY_SECRET_TOKEN pulumi up --stack prod
 ```
 
 I get:
 
 ```
panic: fatal: An assertion has failed

goroutine 194 [running]:
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
private/tmp/pulumi-20221109-4783-xtvj6j/sdk/go/common/util/contract/failfast.go:23
github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.Assert(...)
/private/tmp/pulumi-20221109-4783-xtvj6j/sdk/go/common/util/contract/assert.go:26
github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin.(*provider).Construct(0xc0006e05a0, {{0xc00029d8d8, 0x3}, {0xc00065ffd4, 0x4}, 0xc0009f01b0, {0x0, 0x0, 0x0}, 0x1, ...}, ...)
/private/tmp/pulumi-20221109-4783-xtvj6j/sdk/go/common/resource/plugin/provider_plugin.go:1201 +0x11b4
github.com/pulumi/pulumi/pkg/v3/resource/deploy.(*resmon).RegisterResource(0xc00063f1e0, {0xc00149b858?, 0x1003d96f5?}, 0xc0015221a0)
/private/tmp/pulumi-20221109-4783-xtvj6j/pkg/resource/deploy/source_eval.go:1201 +0x27b9
github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceMonitor_RegisterResource_Handler.func1({0x10249a718, 0xc0006ed5f0}, {0x101e8ea20?, 0xc0015221a0})
/private/tmp/pulumi-20221109-4783-xtvj6j/sdk/proto/go/resource.pb.go:1713 +0x78
github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x10249a718, 0xc0006ed3e0}, {0x101e8ea20, 0xc0015221a0}, 0xc0006e8160, 0xc001500c30)
/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x3f9
github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceMonitor_RegisterResource_Handler({0x101dbe3a0?, 0xc00063f1e0}, {0x10249a718, 0xc0006ed3e0}, 0xc001526000, 0xc0000a7220)
/private/tmp/pulumi-20221109-4783-xtvj6j/sdk/proto/go/resource.pb.go:1715 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000b71860, {0x1024a4c20, 0xc00023c340}, 0xc00151ec60, 0xc0009f0480, 0x1034b3580, 0x0)
/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:1301 +0xb2b
google.golang.org/grpc.(*Server).handleStream(0xc000b71860, {0x1024a4c20, 0xc00023c340}, 0xc00151ec60, 0x0)
/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:1642 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:938 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/google.golang.org/grpc@v1.49.0/server.go:936 +0x28a
```

Here is the code the replicates it:

```typescript
import * as digitalocean from "@pulumi/digitalocean";
import * as kubernetes from "@pulumi/kubernetes";
import * as certmanager from "@pulumi/kubernetes-cert-manager";

const cluster = new digitalocean.KubernetesCluster("do-cluster", {
    region: digitalocean.Region.LON1,
    version: digitalocean.getKubernetesVersions().then(p => p.latestVersion),
    nodePool: {
        name: "default",
        size: digitalocean.DropletSlug.DropletS2VCPU2GB,
        nodeCount: 2,
    },
});

export const kubeconfig = cluster.status.apply(status => {
    if (status === "running") {
        const clusterDataSource = cluster.name.apply(name => digitalocean.getKubernetesCluster({ name }));
        return clusterDataSource.kubeConfigs[0].rawConfig;
    } else {
        return cluster.kubeConfigs[0].rawConfig;
    }
});

const provider = new kubernetes.Provider("do-k8s", { kubeconfig });

const certManagerNamespace = new kubernetes.core.v1.Namespace("do-cert-manager-namespace", {
    metadata: { name: 'cert-manager' },
}, { provider });

const certManager = new certmanager.CertManager("do-cert-manager", {
    installCRDs: true,
    helmOptions: {
        namespace: certManagerNamespace.metadata.name,
    },
}, { provider });
```

I am running it on my MacBook pro, pulumi cli version `v3.46.1`. It has worked in the past, but I have since manually destroyed and want to recreate it.

If I remove the following:

```typescript
const certManager = new certmanager.CertManager("do-cert-manager", {
    installCRDs: true,
    helmOptions: {
        namespace: certManagerNamespace.metadata.name,
    },
}, { provider });
```

it no longer crashes. I am not sure what is going on.

Any ideas?

Should I open a ticket in the digital ocean repo?

Thanks

--- a/changelog/pending/20230921--sdk-go--ensure-assets-of-assetarchive-are-non-nil-when-creating-serializing-and-deserializing.yaml
+++ b/changelog/pending/20230921--sdk-go--ensure-assets-of-assetarchive-are-non-nil-when-creating-serializing-and-deserializing.yaml
@@ -1,0 +1,4 @@
+changes:
+- type: fix
+  scope: sdk/go
+  description: Ensure Assets of AssetArchive are non-nil when creating and deserializing

--- a/sdk/go/common/resource/asset.go
+++ b/sdk/go/common/resource/asset.go
@@ -471,6 +471,11 @@ const (
 )
 
 func NewAssetArchive(assets map[string]interface{}) (*Archive, error) {
+	if assets == nil {
+		// when provided assets are nil, create an empty archive
+		assets = make(map[string]interface{})
+	}
+
 	// Ensure all elements are either assets or archives.
 	for _, asset := range assets {
 		switch t := asset.(type) {
@@ -486,12 +491,18 @@ func NewAssetArchive(assets map[string]interface{}) (*Archive, error) {
 }
 
 func NewPathArchive(path string) (*Archive, error) {
+	if path == "" {
+		return nil, errors.New("path cannot be empty when constructing a path archive")
+	}
 	a := &Archive{Sig: ArchiveSig, Path: path}
 	err := a.EnsureHash()
 	return a, err
 }
 
 func NewURIArchive(uri string) (*Archive, error) {
+	if uri == "" {
+		return nil, errors.New("uri cannot be empty when constructing a URI archive")
+	}
 	a := &Archive{Sig: ArchiveSig, URI: uri}
 	err := a.EnsureHash()
 	return a, err
@@ -603,60 +614,71 @@ func DeserializeArchive(obj map[string]interface{}) (*Archive, bool, error) {
 		hash = h
 	}
 
-	var assets map[string]interface{}
-	if v, has := obj[ArchiveAssetsProperty]; has {
-		assets = make(map[string]interface{})
-		if v != nil {
-			m, ok := v.(map[string]interface{})
-			if !ok {
-				return &Archive{}, false, fmt.Errorf("unexpected archive contents of type %T", v)
-			}
+	if path, has := obj[ArchivePathProperty]; has {
+		pathValue, ok := path.(string)
+		if !ok {
+			return &Archive{}, false, fmt.Errorf("unexpected archive path of type %T", path)
+		}
 
-			for k, elem := range m {
-				switch t := elem.(type) {
-				case *Asset:
-					assets[k] = t
-				case *Archive:
-					assets[k] = t
-				case map[string]interface{}:
-					a, isa, err := DeserializeAsset(t)
+		if pathValue != "" {
+			pathArchive := &Archive{Sig: ArchiveSig, Path: pathValue, Hash: hash}
+			return pathArchive, true, nil
+		}
+	}
+
+	if uri, has := obj[ArchiveURIProperty]; has {
+		uriValue, ok := uri.(string)
+		if !ok {
+			return &Archive{}, false, fmt.Errorf("unexpected archive URI of type %T", uri)
+		}
+
+		if uriValue != "" {
+			uriArchive := &Archive{Sig: ArchiveSig, URI: uriValue, Hash: hash}
+			return uriArchive, true, nil
+		}
+	}
+
+	if assetsMap, has := obj[ArchiveAssetsProperty]; has {
+		m, ok := assetsMap.(map[string]interface{})
+		if !ok {
+			return &Archive{}, false, fmt.Errorf("unexpected archive contents of type %T", assetsMap)
+		}
+
+		assets := make(map[string]interface{})
+		for k, elem := range m {
+			switch t := elem.(type) {
+			case *Asset:
+				assets[k] = t
+			case *Archive:
+				assets[k] = t
+			case map[string]interface{}:
+				a, isa, err := DeserializeAsset(t)
+				if err != nil {
+					return &Archive{}, false, err
+				} else if isa {
+					assets[k] = a
+				} else {
+					arch, isarch, err := DeserializeArchive(t)
 					if err != nil {
 						return &Archive{}, false, err
-					} else if isa {
-						assets[k] = a
-					} else {
-						arch, isarch, err := DeserializeArchive(t)
-						if err != nil {
-							return &Archive{}, false, err
-						} else if !isarch {
-							return &Archive{}, false, fmt.Errorf("archive member '%v' is not an asset or archive", k)
-						}
-						assets[k] = arch
+					} else if !isarch {
+						return &Archive{}, false, fmt.Errorf("archive member '%v' is not an asset or archive", k)
 					}
-				default:
-					return &Archive{}, false, fmt.Errorf("archive member '%v' is not an asset or archive", k)
+					assets[k] = arch
 				}
+			default:
+				return &Archive{}, false, fmt.Errorf("archive member '%v' is not an asset or archive", k)
 			}
 		}
-	}
-	var path string
-	if v, has := obj[ArchivePathProperty]; has {
-		p, ok := v.(string)
-		if !ok {
-			return &Archive{}, false, fmt.Errorf("unexpected archive path of type %T", v)
-		}
-		path = p
-	}
-	var uri string
-	if v, has := obj[ArchiveURIProperty]; has {
-		u, ok := v.(string)
-		if !ok {
-			return &Archive{}, false, fmt.Errorf("unexpected archive URI of type %T", v)
-		}
-		uri = u
+
+		assetArchive := &Archive{Sig: ArchiveSig, Assets: assets, Hash: hash}
+		return assetArchive, true, nil
 	}
 
-	return &Archive{Hash: hash, Assets: assets, Path: path, URI: uri}, true, nil
+	// if we reached here, it means the archive is empty,
+	// we didn't find a non-zero path, non-zero uri nor non-nil assets
+	// we will consider this to be an empty assets archive then with zero assets
+	return &Archive{Sig: ArchiveSig, Assets: make(map[string]interface{}), Hash: hash}, true, nil
 }
 
 // HasContents indicates whether or not an archive's contents can be read.

--- a/sdk/go/common/resource/asset_test.go
+++ b/sdk/go/common/resource/asset_test.go
@@ -27,6 +27,8 @@ import (
 	"runtime"
 	"testing"
 
+	"github.com/stretchr/testify/require"
+
 	"github.com/stretchr/testify/assert"
 
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
@@ -537,6 +539,19 @@ func TestFileExtentionSniffing(t *testing.T) {
 	assert.Equal(t, ArchiveFormat(TarGZIPArchive), detectArchiveFormat("./some/path/my.file.tgz"))
 	assert.Equal(t, ArchiveFormat(JARArchive), detectArchiveFormat("./some/path/my.file.jar"))
 	assert.Equal(t, ArchiveFormat(NotArchive), detectArchiveFormat("./some/path/who.even.knows"))
+}
+
+func TestEmptyArchiveRoundTrip(t *testing.T) {
+	t.Parallel()
+
+	emptyArchive, err := NewAssetArchive(nil)
+	require.NoError(t, err, "Creating an empty archive should work")
+	assert.True(t, emptyArchive.IsAssets(), "even empty archives should be have empty assets")
+	serialized := emptyArchive.Serialize()
+	deserialized, ok, err := DeserializeArchive(serialized)
+	assert.NoError(t, err, "Deserializing an empty archive should work")
+	assert.True(t, ok, "Deserializing an empty archive should return true")
+	assert.True(t, deserialized.IsAssets(), "Deserialized archive should be an AssetsArchive")
 }
 
 func TestInvalidPathArchive(t *testing.T) {

